source/MaterialXGenShader/ShaderNode.h

```diff
@@ -144,9 +144,16 @@ class ShaderInput : public ShaderPort
     /// Get optional channels value
     const string& getChannels() const { return _channels; }
 
+    /// Set isBindInput value. If true, either a BindInput or BindParam on a shaderref
+    void setIsBindInput(bool isBindInput) { _isBindInput = isBindInput; }
+
+    /// Get isBindInput value
+    const bool& getIsBindInput() const { return _isBindInput;  }
+
   protected:
     ShaderOutput* _connection;
     string _channels;
+    bool _isBindInput;
```

Comment:
I found a more generic way to cache this. 

On the `ShaderPort` class there is an `_flags` data member. We can reuse this by adding in a new flag, say  BIND_INPUT. Note that there is already an EMITTED flag so that value cannot be reused.

The condition test would then be something like `input.getFlags() & ShaderInput:BIND_INPUT` and setting is `input.setFlags(input.getFlags() | ShaderInput::BIND_INPUT)`.